# Assignment 6.7 — Fibonacci Sequence

Your challenge is to write a Python function big_fibonacci that takes a number n and returns the first number of the Fibonacci sequence that has n digits. For instance, 1 is the first number with 1 digit in the sequence, 13 is the first number with 2 digits in the sequence, and 144 is the first number with 3 digits in the sequence.

In [254]:
def big_fibonacci(n):
    if n==1:
        return 1
    a=1
    b=1
    c=a+b
    while len(str(c))<n:
        a=b
        b=c
        c=a+b
    return c
#A return statement is used to end the execution of the function call and “returns” the result (value of the expression following the return keyword) to the caller. 
#The statements after the return statements are not executed.

In [255]:
big_fibonacci(1)

1

In [256]:
big_fibonacci(2)

144

In [253]:
big_fibonacci(10)

1134903170

# Assignment 6.8 — FizzBuzz

Write a Python script called fizzbuzz.py that prints something for each number n from 1 to 1000, according to the following rules:

* If n is divisible by 3, print "Fizz".
* If n is divisible by 5, print "Buzz".
* If n is divisible by both 3 and 5, print "FizzBuzz".
* If n is not divisible by either 3 or 5, just print the number n.
So, your program should print out 1000 lines of text in total, and the first 20 lines should look like this:

In [285]:
n=0
while 0<=n<=999:
    n=n+1
    if n%3==0 and n%5==0:
        print ('FizzBuzz')
    elif n%3==0 and n%5!=0:
        print('Fizz')
    elif n%3!=0 and n%5==0:
        print('Buzz')
    else:
        print(n)

1
2
Fizz
4
Buzz
Fizz
7
8
Fizz
Buzz
11
Fizz
13
14
FizzBuzz
16
17
Fizz
19
Buzz
Fizz
22
23
Fizz
Buzz
26
Fizz
28
29
FizzBuzz
31
32
Fizz
34
Buzz
Fizz
37
38
Fizz
Buzz
41
Fizz
43
44
FizzBuzz
46
47
Fizz
49
Buzz
Fizz
52
53
Fizz
Buzz
56
Fizz
58
59
FizzBuzz
61
62
Fizz
64
Buzz
Fizz
67
68
Fizz
Buzz
71
Fizz
73
74
FizzBuzz
76
77
Fizz
79
Buzz
Fizz
82
83
Fizz
Buzz
86
Fizz
88
89
FizzBuzz
91
92
Fizz
94
Buzz
Fizz
97
98
Fizz
Buzz
101
Fizz
103
104
FizzBuzz
106
107
Fizz
109
Buzz
Fizz
112
113
Fizz
Buzz
116
Fizz
118
119
FizzBuzz
121
122
Fizz
124
Buzz
Fizz
127
128
Fizz
Buzz
131
Fizz
133
134
FizzBuzz
136
137
Fizz
139
Buzz
Fizz
142
143
Fizz
Buzz
146
Fizz
148
149
FizzBuzz
151
152
Fizz
154
Buzz
Fizz
157
158
Fizz
Buzz
161
Fizz
163
164
FizzBuzz
166
167
Fizz
169
Buzz
Fizz
172
173
Fizz
Buzz
176
Fizz
178
179
FizzBuzz
181
182
Fizz
184
Buzz
Fizz
187
188
Fizz
Buzz
191
Fizz
193
194
FizzBuzz
196
197
Fizz
199
Buzz
Fizz
202
203
Fizz
Buzz
206
Fizz
208
209
FizzBuzz
211
212
Fizz
214
Buzz
Fizz
217
218
Fizz
Buzz
221
Fizz
223
224
Fi

# Assignment 6.9 — Prime Number Test

In [260]:
def is_prime(n):
    if n<=1:
        print('Number has to be greater than 1 to be prime!')
    else:
        i=1
        while i<n-1:
            i=i+1
            result=n%i
            if result==0:
                return False
        else:
            return True

In [268]:
print(is_prime(2))
print(is_prime(6))
print(is_prime(8))
print(is_prime(17))
print(is_prime(23))
print(is_prime(25))

True
False
False
True
True
False


# Assignment 6.10 — Find a Lot of Prime Numbers

Write a Python function called primes_below that takes a number n and returns a list of all prime numbers less than n.

In [279]:
def primes_below(n):
    output=[]
    i=1
    while i+1<n:
        i=i+1
        if is_prime(i)==True:
            output.append(i)
    return output

In [282]:
primes_below(17)

[2, 3, 5, 7, 11, 13]

In [283]:
primes_below(8)

[2, 3, 5, 7]

In [284]:
primes_below(2)

[]

# Assignment 6.11 — Find Palindrome Primes

A number (or word) is a palindrome if it looks the same forwards or backwards. For instance, the number 12321 is a palindrome, because if you reverse its digits you get the same number.
* Here are all of the three-digit primes which are also palindromes, packed in a Python list.
* [101, 131, 151, 181, 191, 313, 353, 373, 383, 727, 757, 787, 797, 919, 929]
* Your task is to create a Python program called palindrome_primes.py that creates a list of all of the five-digit prime numbers which are palindromes. Your program can do many steps, but it should only print the result list.

In [301]:
output=[]
n=10000
while len(str(n))==5:
    n=n+1
    if is_prime(n)==True:
        if str(n)[0]==str(n)[4] and str(n)[1]==str(n)[3]:
            output.append(n)
print(output)

[10301, 10501, 10601, 11311, 11411, 12421, 12721, 12821, 13331, 13831, 13931, 14341, 14741, 15451, 15551, 16061, 16361, 16561, 16661, 17471, 17971, 18181, 18481, 19391, 19891, 19991, 30103, 30203, 30403, 30703, 30803, 31013, 31513, 32323, 32423, 33533, 34543, 34843, 35053, 35153, 35353, 35753, 36263, 36563, 37273, 37573, 38083, 38183, 38783, 39293, 70207, 70507, 70607, 71317, 71917, 72227, 72727, 73037, 73237, 73637, 74047, 74747, 75557, 76367, 76667, 77377, 77477, 77977, 78487, 78787, 78887, 79397, 79697, 79997, 90709, 91019, 93139, 93239, 93739, 94049, 94349, 94649, 94849, 94949, 95959, 96269, 96469, 96769, 97379, 97579, 97879, 98389, 98689]


In [300]:
#3 digit version
output=[]
n=100
while len(str(n))==3:
    n=n+1
    if is_prime(n)==True:
        if str(n)[0]==str(n)[2]:
            output.append(n)
print(output)

[101, 131, 151, 181, 191, 313, 353, 373, 383, 727, 757, 787, 797, 919, 929]


# Assignment 6.13 — Triangle Class

* Create a Triangle class. Provide three numbers, giving the length of each side. 
* Your rectangle class should have perimeter, area, and scale methods in analogy with the same methods on the Rectangle class.
* Additionally, add two more methods to the Triangle class. First, implement a is_valid method so that t.is_valid() returns True if t is a valid triangle, or False otherwise. For example, there is no triangle with side lengths 1, 3, and 5. 
* Next, implement a is_right method that returns True if the triangle is a right triangle (having a 90 degree angle) and False otherwise. 

In [303]:
import math as m

In [430]:
class Triangle():
    def __init__(self,a,b,c):
        self.length1=a
        self.length2=b
        self.length3=c

    def perimeter(self):
        return self.length1+self.length2+self.length3

    def area(self):
        return m.sqrt((0.5*self.perimeter())*((0.5*self.perimeter())-self.length1)*((0.5*self.perimeter())-self.length1)*((0.5*self.perimeter())-self.length1))
            
    def scale(self,scale_factor):
        return Triangle(scale_factor*self.length1, scale_factor*self.length2,scale_factor*self.length3)

    def is_valid(self):
        #the sum of 2 sides of a triangle must be greater than the third side
        if (self.length1<(self.length2+self.length3)) and (self.length2<(self.length1+self.length3)) and (self.length3<(self.length1+self.length2)):
            return True
        else:
            return False
        
    #def is_valid2(self):
        #if (self.length1>=(self.length2+self.length3)) or (self.length2>=(self.length1+self.length3)) or (self.length3>=(self.length1+self.length2)):
            #return False
        #else:
            #return True
    
    def is_right(self):
        #Pythagorean theorem
        if self.length1==m.sqrt((self.length2)**2+(self.length3)**2) or self.length2==m.sqrt((self.length1)**2+(self.length3)**2) or self.length3==m.sqrt((self.length1)**2+(self.length2)**2):
            return True
        else:
            return False

In [426]:
t = Triangle(8,6,10)

In [428]:
print('Does a triangle with side lengths {}, {}, {} exist? {}'.format(t.length1,t.length2,t.length3,t.is_valid()))

Does a triangle with side lengths 8, 6, 10 exist? True


In [411]:
print('Perimeter equals %d.' % t.perimeter())
print('Area equals %d.' % t.area())

Perimeter equals 24.
Area equals 27.


In [412]:
q=t.scale(2)

In [413]:
print(q.length1, q.length2,q.length3)

16 12 20


In [429]:
print('Is the triangle with side lengths {}, {}, {} a right one? {}'.format(t.length1,t.length2,t.length3,t.is_right()))

Is the triangle with side lengths 8, 6, 10 a right one? True


In [431]:
z = Triangle  (7,1,3)

In [432]:
print('Does a triangle with side lengths {}, {}, {} exist? {}'.format(z.length1,z.length2,z.length3,z.is_valid()))

Does a triangle with side lengths 7, 1, 3 exist? False
